### ISSUE TITLE
regression with scatter and intel19

### ISSUE THREAD

**Issue by aminiussi** [boostorg/mpi#76]
```
capture-output ../../../bin.v2/libs/mpi/test/scatter_test-7.test/intel-linux-std11/debug/visibility-hidden/scatter_test-7-run
====== BEGIN OUTPUT ======
Scattering integers from root 0...
Scattering integers from root 1...
Scattering integers from root 2...
Scattering integers from root 3...
Scattering integers from root 4...
Scattering integers from root 5...
Scattering integers from root 6...
Scattering GPS positions from root 0...
Scattering GPS positions from root 1...
Scattering GPS positions from root 2...
Scattering GPS positions from root 3...
Scattering GPS positions from root 4...
Scattering GPS positions from root 5...
Scattering GPS positions from root 6...
Scattering string from root 0...
Scattering string from root 1...
../../../boost/test/minimal.hpp(136): exception "memory access violation at address: 0x75d631c3: no mapping at fault address" caught in function: 'int main(int, char **)'
```

It seems that process zero get a slightly shifted archive buffer, just enough to screw the size wrong.

Issues 03708514 was file to Intel.

**Comment by aminiussi** [comment]
A minimal test case, withou Boost.MPI involved, was provided to Intel support.
[plain_cpp.txt](https://github.com/boostorg/mpi/files/2535186/plain_cpp.txt)

**Comment by aminiussi** [comment]
Intel agreed there is a bug and provided possible workaround:
[alainm@pollux test]$ export  I_MPI_ADJUST_SCATTERV=1

**Comment by aminiussi** [comment]
Fixed with intel 2019.4
